### PR TITLE
Add elastic test

### PR DESCRIPTION
**Pull request by olukas**

### CODE REVIEW

File: elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        Collections.sort(list);
+        return list;
+    }
+
+    private void assertEmptyResults(JetInstance client) {
+        IList<String> list = client.getList(SINK_LIST_NAME);
+        assertTrue(list.isEmpty());
+    }
+
+    private void clearSinkMap(JetInstance client) {
```
Comment (frant-hartm) [inline elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:186]: Should be `clearSinkList`

File: elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                .writeTo(elasticSink);
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.setName("ElasticTest_writeTo_" + indexCounter);
+        client.newJob(toElastic, jobConfig).join();
+    }
+
+    private void executeReadFromElasticPipeline(JetInstance client, int indexCounter) {
+        final String ip = elasticIp;
+        BatchSource elasticSource = ElasticSources.elastic(
```
Comment (frant-hartm) [inline elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:125]: Should be able to use `BatchSource<String>`

File: elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        client.newJob(toElastic, jobConfig).join();
+    }
+
+    private void assertResults(JetInstance client, int indexCounter) {
+        IList<String> list = client.getList(SINK_LIST_NAME);
+        assertEquals(itemCount, list.size());
+        List<String> copiedList = new ArrayList<>(list);
+        Collections.sort(copiedList);
+        List<String> expectedList = prepareExpectedList(indexCounter);
+        assertEquals(expectedList, copiedList);
```
Comment (gurbuzali) [inline elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:169]:
```suggestion
        IList<String> list = client.getList(SINK_LIST_NAME);
        Set<String> set = new HashSet<>();
        String prefix = "_index-" + indexCounter + "_";
        for(String item: list) {
            assertTrue(item.startsWith(prefix));
            set.add(item);
        }
        assertEquals(itemCount, set.size());
```
This assertion is very costly: we are sorting the IList then create a test list which creates a lot of Strings for each assertion. We don't need to check explicitly for each item I think. asserting that there is no missing or duplicate item and that each item starts with the mentioned index counter should be enough.

Comment (gurbuzali) [reply elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:169]: we can get rid off `prepareExpectedList` if we apply this

File: elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    @Override
+    public void test(JetInstance client, String name) throws Exception {
+        logger.info("Elastic ip: " + elasticIp);
+        clearSinkList(client);
+        int jobCounter = 0;
+        long begin = System.currentTimeMillis();
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            clearSinkList(client);
+
+            int indexCounter = jobCounter;
```
Comment (gurbuzali) [inline elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:80]: `indexCounter` is always equals to jobCounter, we can get rid off it

File: elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+            executeReadFromElasticPipeline(client, indexCounter);
+
+            assertEmptyResults(client);
+
+            if (jobCounter % LOG_JOB_COUNT_THRESHOLD == 0) {
+                logger.info("Job count: " + jobCounter);
+            }
+
+            jobCounter++;
+            sleepSeconds(SLEEP_BETWEEN_TABLE_READS_SECONDS);
```
Comment (gurbuzali) [inline elastic-test/src/main/java/com/hazelcast/jet/tests/elastic/ElasticTest.java:98]: we have 5 sec wait between jobs and we log for every 100 jobs. this makes it like every 8 minutes we log the counter. I think we should wither lower the wait time or the logging frequency (latter probably).

**Pull request merged**